espflash/src/flasher.rs

```diff
@@ -138,6 +138,16 @@ struct BeginParams {
     offset: u32,
 }
 
+#[derive(Zeroable, Pod, Copy, Clone, Debug)]
+#[repr(C)]
+struct BeginParamsS2 {
```

Comment:
This `encrypted` field applies to all chips except the ESP8266/ESP32 as far as I know. This should probably be the default struct, and we can have a `BeginParamsLegacy` or something to that effect instead.

Reply:
I was thinking of having some way of having per-chip command/parameter structs or otherwise moving some of the flashing logic to the chips.